open-api/rest-catalog-open-api.yaml

```diff
@@ -3129,6 +3204,11 @@ components:
          - `s3.secret-access-key`: secret for credentials that provide access to data in S3 
          - `s3.session-token`: if present, this value should be used for as the session token 
          - `s3.remote-signing-enabled`: if `true` remote signing should be performed as described in the `s3-signer-open-api.yaml` specification
+
+        ## Credentials
+
+        Credentials for ADLS / GCS / S3 are provided through the `credentials` field. Clients should first check whether the
```

Comment:
Would it make sense to take it one step further and have clients request credentials for narrower use cases?.. For example separating read and write access.

To be clear, I mean adding a separate REST endpoint for obtaining storage credentials.

Reply:
Good point!
Obtaining credentials every time a table's loaded is often just overhead. Just inspecting table-metadata contents doesn't need any object-store credentials.
Also if there's finally a way to refresh credentials, the lifetime of those can be shortened.

There's a lot of room for improvement here

Reply:
Re: refresh: it should even be possible to support very long client sessions too, where credentials are rotated during the same query: initial access is based on old credentials, overlap in the middle (both sets of credentials valid) and using new credentials at the end of processing.

Reply:
@nastra ; Great to hear that this kind of improvement is in the works :) I assume a dev list discussion will start as soon as that change can be shared.

Reply:
ok. thx!